docs/website/contents/for-developers/ProtocolVersioning.md

```diff
@@ -0,0 +1,155 @@
+
+### Precise Answer
+
+The protocol version must be incremented if an old node and a new node would otherwise disagree on any of the following.
+
+- *Chain Validity Predicates*, the validity predicates within the Message Validity Rung for block headers and blocks.
+- *Chain Order*, the [partial order](https://en.wikipedia.org/wiki/Partially_ordered_set) within the Message Validity Rung that determines one chain is better than another.
+  Tiebreakers are a gray area, since Praos does not inherently rely on them.
+- *Chain Data Model*, the parts of the Message Data Rung used to define the Chain Validity Predicates and the Chain Order.
+    - A change to the data model of the node's maintained state --- as opposed to its messages --- would not inherently trigger this.
+    - However, if such a change were meaningful (ie not just a refactor/optimization/etc), then some Chain Validity Predicates would necessarily also change.
+- *Chain Codecs*, the parts of the mapping from the CBOR Rung to the Chain Data Model that affect the hashes therein.
+  Two nodes won't even be able to exchange transactions, block headers, and/or blocks if they disagree on their identifying hashes.
+
+The most common change in Cardano's protocol so far has been the addition of a new kind/variant of transaction, which triggers all of these except the Chain Order.
+The Chain Order has only changed from Praos's length to Ouroboros Genesis's length-soon-after-the-intersection, which is slight.
+On the other hand, the tiebreaker has changed, and out of caution those changes were bundled up with unrelated changes that independently required a protocol version increment.
+
+It's also worth noting that the Consensus Layer and Network Layer of the Message Data Rung can change without incrementing the protocol version.
+The most common example is that queries can be changed without triggering any of the above, since queries do not contain any of the Chain Data Model.
+Another example is that the structure of the Consensus envelope could change.
+Or the mini protocols --- excluding their payload --- could change.
+All such changes are conditional on the per-connection version negotiated by the Handshake mini protocol instead of the protocol version.
+
+## Ledger Eras
+
+Any Cardano node implemented in a strongly-typed language is likely going to declare distinct types for Chain Data Models that differ between protocol versions; more precise types yield stronger assurances from the typechecker.
+For the sake of such implementations, the ledger introduces a new era whenever the Chain Data Model changes.
+The [sequence of eras](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0059/feature-table.md) at the time of writing are: Byron [^byron-halves], Shelley, Allegra, Mary, Alonzo, Babbage, and Conway.
+
+A Cardano node implementation or other tooling implemented in a language prone to dynamic typing might find it convenient to ignore the era tags for the most part.
```

Comment:
I appreciate the distinction between strongly-typed nodes and other nodes. Perhaps this could end up in the Blueprint

Reply:
Great to hear! The blueprint is certainly in the back of my mind as I'm writing this document.